### ISSUE TITLE
Rolling back a disk fails after stopping a the machine

### ISSUE THREAD

**Issue by ashraffouda** [zero-os/0-Disk#405]
Rolling back a disk with this scenario fails 
1- Create a disk 
2- create a vm with this disk 
3- stop the machine 
4- resize the disk 
5- do roll back for this disk 
the error is 
```
 File "/usr/lib/python3.5/concurrent/futures/thread.py", line 55, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/tmp/actions/vdisk/49fc5356ac3092fc5e7e040537449d5e.py", line 225, in rollback
    raise j.exceptions.RuntimeError("Failed to run zeroctl restore {} {}".format(result.stdout, result.stderr))
JumpScale9.errorhandling.JSExceptions.RuntimeError: ERROR: Failed to run zeroctl restore nil last hash Error: nil last hash
Usage:
  zeroctl restore vdisk id [flags]

Flags:
      --config SourceConfig        config resource: dialstrings (etcd cluster) or path (yaml file) (default config.yml)
      --end-timestamp uint         end timestamp in nanosecond(default 0: until the end)
  -f, --force                      when given delete the vdisk if it already existed
  -h, --help                       help for vdisk
      --k int                      K variable of erasure encoding (default 4)
      --m int                      M variable of erasure encoding (default 2)
      --nonce string               hex nonce used for encryption (default "37b8e8a308c354048d245f6d")
      --priv-key string            private key (default "12345678901234567890123456789012")
      --start-timestamp uint       start timestamp in nanosecond(default 0: since beginning)
      --storage-addresses string   comma seperated list of redis compatible connectionstrings (format: '<ip>:<port>[@<db>]', eg: 'localhost:16379,localhost:6379@2'), if given, these are used for all vdisks, ignoring the given confi
g

Global Flags:
  -v, --verbose   log available information

 ((type:runtime.error)
[Wed16 13:52] - Job.py              :56  :j.core.jobcontroller.job.35e18ec418e72c0f437722c5e1e20e0c - ERROR    - Traceback (most recent call last):
  File "/usr/lib/python3.5/concurrent/futures/thread.py", line 55, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/tmp/actions/vdisk/49fc5356ac3092fc5e7e040537449d5e.py", line 346, in processChange
    j.tools.async.wrappers.sync(service.executeAction('rollback', args={'timestamp': args['timestamp']}, context=job.context))
  File "/opt/code/github/jumpscale/lib9/JumpScale9Lib/tools/async/Wrappers.py", line 12, in sync
    return loop.run_until_complete(coro)
  File "/usr/lib/python3.5/asyncio/base_events.py", line 387, in run_until_complete
    return future.result()
  File "/usr/lib/python3.5/asyncio/futures.py", line 274, in result
    raise self._exception
  File "/usr/lib/python3.5/asyncio/tasks.py", line 241, in _step
    result = coro.throw(exc)
  File "/opt/code/github/jumpscale/ays9/JumpScale9AYS/ays/lib/Service.py", line 723, in executeAction
    return await self.executeActionJob(action, args, context=context)
  File "/opt/code/github/jumpscale/ays9/JumpScale9AYS/ays/lib/Service.py", line 748, in executeActionJob
    result = await job.execute()
  File "/usr/lib/python3.5/asyncio/futures.py", line 361, in __iter__
    yield self  # This tells Task to wait for completion.
  File "/usr/lib/python3.5/asyncio/tasks.py", line 296, in _wakeup
    future.result()
  File "/usr/lib/python3.5/asyncio/futures.py", line 274, in result
    raise self._exception
  File "/usr/lib/python3.5/concurrent/futures/thread.py", line 55, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/tmp/actions/vdisk/49fc5356ac3092fc5e7e040537449d5e.py", line 225, in rollback
    raise j.exceptions.RuntimeError("Failed to run zeroctl restore {} {}".format(result.stdout, result.stderr))
JumpScale9.errorhandling.JSExceptions.RuntimeError: ERROR: Failed to run zeroctl restore nil last hash Error: nil last hash
Usage:
  zeroctl restore vdisk id [flags]
```

**Comment by iwanbk** [comment]
> umpScale9.errorhandling.JSExceptions.RuntimeError: ERROR: Failed to run zeroctl restore nil last hash Error: nil last hash

the restorer couldn't find the tlog's last hash. Could you share more info about tlog related config?

**Comment by ashraffouda** [comment]
this is how I created the tlog 
```
{
  "label": "tl",
  "servers": 2,
  "driveType": "ssd",
  "clusterType": "tlog",
  "k":1,
"m":1,
  "nodes": ["248a07e777e0"]
}
```
and this how we save the config for tlog and other stuff
https://github.com/zero-os/0-orchestrator/blob/master/templates/vdisk/actions.py#L136

**Comment by GlenDC** [comment]
It would be very helpful if you could also give the configuration at step of your process, @ashraffouda, as we never use 0-Orchestrator. Linking some JSON config for 0-orchestrator here isn't really going to aid in helping you I'm afraid.

**Comment by iwanbk** [comment]
@GlenDC more info about the problem. Had a chat with @ashraffouda, i asked him yesterday to provide the result of 
```
lrange last_hash_vdisk-id 0 5
```
on one of his tlog's ardb server, he didn't found the key exist, and only hash `1) "lba:disk2"` as key.

Then i told him to use `tlogCluster` as described in https://github.com/zero-os/0-Disk/blob/master/docs/config.md#reading-data-1

Hope he can check it today.

**Comment by GlenDC** [comment]
@ashraffouda and @iwanbk did you get any further with this issue, as I haven't seen any progress in the last 11 days...

**Comment by GlenDC** [comment]
Also @ashraffouda make sure to retry your same progress again, with the latest master code, as the tlogserver is now using 0-stor, which might for whatever reason have solved your bug as well.

**Comment by GlenDC** [comment]
Once this issue is resolved make sure to open a PR with the solution using [the 1.1.0-alpha-8 branch](https://github.com/zero-os/0-Disk/tree/1.1.0-alpha-7) as the base branch. Master is from now on meant for beta code.

**Comment by iwanbk** [comment]
I think is it a duplicate of #425 or outdated

@khaledkbadr @ashraffouda  please close or provide more info

**Comment by GlenDC** [comment]
I think they can only confirm that (it is a duplicate), once you have a fix for #425, no?

**Comment by iwanbk** [comment]
> I think they can only confirm that (it is a duplicate), once you have a fix for #425, no?

perhaps.
I though it is duplicate because @khaledkbadr said that he took over @ashraffouda work

**Comment by GlenDC** [comment]
Sure, but that's just guessing work. Best is to wait until we have a confirmed fix for #425 which also resolves this issue.

**Comment by GlenDC** [comment]
You were right, it turns out that this is indeed a duplicate issue of #425.
